WordPress/gutenberg
# Block Moving Mode: Prevent the block from being moved into itself

Fixes: #54016

## What?
This PR ensures that when a block is moved in Move to mode, it cannot be moved inside its own block.

## Why?
The block is not supposed to be moved into itself, because if it were, an infinite loop would occur and the browser would freeze.

## How?

When the block itself to be moved is selected, pressing the right key cancels the attempt to enter the block. At the same time, we added basic E2E tests.

### Testing Instructions for Keyboard

- Insert two group blocks with multiple paragraphs.
- Select the first group block.
- Select "Move to" from the block toolbar.
- Press the right key to move it into itself. Nothing should happen.
- Press the down key to move to the next group block.
- Press the right key and it should move correctly into the group block.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/c296bb9c-461d-4181-a2cf-60597e7b310a
